architecture/core.rst

```diff
@@ -430,6 +430,28 @@ Interrupt consumers main API functions are:
     - ``interrupt_remove_handler()`` to unregister an interrupt handler
       function from an interrupt.
 
+When the configuration switch ``CFG_DT`` is enabled, OP-TEE core provides a
+way to describe the platform's interrupt topology using Device Tree.
+When used, the platform must provide a secure device tree blob
+(DTB) which platform and device drivers can use to get references to
+interrupt controllers and relative interrupt numbers they consume.
+The Main API functions are:
+
+    - ``dt_register_interrupt_provider()`` to register an interrupt
+      controller related to a given node in the Device Tree. This function
+      requires, among its arguments, a callback function that is called when
+      an interrupt consumer requires a reference to a specific interrupt
+      exposed by the controller. The callback function shall be of type
+      ``dt_get_itr_func`` and it shall allocate a temporary instance of
+      ``struct irq_desc`` to describe the interrupt. This allocation shall
+      to done using ``malloc()`` or like (``calloc()``, ``memalign()``
+      etc...) and is automatically freed with ``free()``.
+
+    - ``dt_get_interrupt()``, ``dt_get_interrupt_by_index()`` and
```

Comment:
`interrupt_dt_get()`, `interrupt_dt_get_by_index()`, and `interrupt_dt_get_by_name()`?

Reply:
and again and again and again
Thanks.